sdks/java/core/src/main/java/org/apache/beam/sdk/util/FileIOChannelFactory.java

```diff
@@ -36,24 +36,38 @@
 import java.nio.file.FileSystems;
 import java.nio.file.Files;
 import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
 import java.nio.file.PathMatcher;
-import java.nio.file.Paths;
 import java.util.Collection;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.regex.Matcher;
 
 /**
- * Implements IOChannelFactory for local files.
+ * Implements {@link IOChannelFactory} for local files.
  */
 public class FileIOChannelFactory implements IOChannelFactory {
   private static final Logger LOG = LoggerFactory.getLogger(FileIOChannelFactory.class);
 
+  /**
+   *  Converts the given file spec to a java {@link File}. If {@code spec} is actually a URI with
+   *  the {@code file://} prefix, then this function will ensure that the returned {@link File}
+   *  has the correct path.
+   */
+  private static File specToFile(String spec) {
+    // Strip the "file://" prefix from the spec, if present.
+    if (spec.toLowerCase().startsWith("file://")) {
+      return new File(spec.substring("file://".length()));
```

Comment:
+1 to using existing libraries. It is best practice, but in this case also the code here is wrong.

The URL `file:/some/absolute/path` is perfectly valid and has the same meaning as `file:///some/absolute/path`. The former is an omitted authority section and the latter is a present authority section with omitted host.

And anyhow IMO it would read more clearly, using `java.net.URI` just as an example:

```java
    URI uri = new URI(spec)
    if (uri.scheme().equals("file")) {
      return uri.getPath();
    }
```